scripts/Muon/thread_model.py

```diff
@@ -9,27 +11,57 @@ class ThreadModel(QThread):
     A wrapper to allow threading with
     the MaxEnt models.
     """
-    def __init__(self,model):
+    exceptionSignal = QtCore.pyqtSignal(object)
+
+    def __init__(self, model):
         QThread.__init__(self)
-        self.model=model
+        self.model = model
 
     def __del__(self):
         self.wait()
 
     def run(self):
+        self.user_cancel = False
         try:
             self.model.execute()
             self.model.output()
         except KeyboardInterrupt:
             pass
+        except Exception as error:
+            if self.user_cancel:
+                print("User ended job")
+            else:
+                # print("waa ",self.user_cancel,error.__class__.__name__)
+                self.sendSignal(error)
+            pass
+
+    def sendSignal(self, error):
+        self.exceptionSignal.emit(error)
+        # pass
+
+    def join(self):
+        print(self.exception)
```

Comment:
Is this print statement supposed to still be here? If it is might be better to replace it with some logging